components/ResultsPage/YourAnswers.tsx

```diff
@@ -0,0 +1,275 @@
+import { WebTranslations } from '../../i18n/web'
+import { livingCountry as enCountry } from '../../i18n/api/countries/en'
+import { livingCountry as frCountry } from '../../i18n/api/countries/fr'
+import en from '../../i18n/api/en'
+import fr from '../../i18n/api/fr'
+import { Locale } from '../../utils/api/definitions/enums'
+import { FieldKey } from '../../utils/api/definitions/fields'
+
+
+
+export const YourAnswers: React.VFC<{
+  title: string
+  questions: Array<[ string, string ]>
+}> = ({ title, questions }) => {
+  const root = useStore()
+
+  const tsln = useTranslation<WebTranslations>()
+  
+  const currentLocale = useRouter().locale
+  const locale = currentLocale == 'en' ? Locale.EN : Locale.FR
+
+  const answers = questions.filter( question => question[0] !== '_language')
+  const answersKeys = answers.map(([key,_]) => key)
+
+  function getLivingCountry(country: string): {key: string,text: string} {
+    if (country === undefined) return undefined
+    if (currentLocale == 'en') 
+      return enCountry.find((val) => val.key === country )
+    else 
+      return frCountry.find((val) => val.key === country )
+  }
+
+  function getLegalStatus(status: string): {key: string,text: string} {
+    if (status === undefined) return undefined
+    if (currentLocale == 'en') 
+      return en.questionOptions.legalStatus.find((val) => val.key === status )
+    else 
+      return fr.questionOptions.legalStatus.find((val) => val.key === status )
+  }
+
+  function getMaritalStatus(status: string): {key: string,text: string} {
+    if (status === undefined) return undefined
+    if (currentLocale == 'en') 
+      return en.questionOptions.maritalStatus.find((val) => val.key === status )
+    else 
+      return fr.questionOptions.maritalStatus.find((val) => val.key === status )
+  }
+
+  function getPartnerBenefitStatus(status: string): {key: string,text: string} {
+    if (status === undefined) return undefined
+    if (currentLocale == 'en') 
+      return en.questionOptions.partnerBenefitStatus.find((val) => val.key === status )
+    else 
+      return fr.questionOptions.partnerBenefitStatus.find((val) => val.key === status )
+  }
+
+  
+
+  if (answersKeys.length === 0) 
+  return (
+    <div className="fz-10">
+      <div className="p-8 bg-emphasis rounded mt-8 md:mt-0 md:max-w-[380px]">
+        <h3 className="h3">{title}</h3>
+        <div className='py-4'>
+          No answers found
+        </div>
+      </div>
+    </div>
+  )
+
+  console.log('anskeys = ', answersKeys)
+
+  if (answersKeys.length !== 0) 
+  return (
+    <div className="fz-10">
+      <div className="p-8 bg-emphasis rounded mt-8 md:mt-0 md:max-w-[380px]">
+        <h3 className="h3">{title}</h3>
+        
+        {/* Age */}
+        {(answersKeys.indexOf(FieldKey.AGE) >= 0 &&
+          <div className='py-4 border-b-2 border-info-border'>
+            {tsln.resultsQuestions[answers[answersKeys.indexOf(FieldKey.AGE)][0]]} <br />
```

Comment:
This file should not have any reference to any specific questions/fields. We keep all field logic/definitions in the backend, and the frontend receives it and processes it. This is so that when we make changes to fields, it only has to be in one place.

So, please refactor this to use loops rather than explicitly defining every single question. You can see here for how we implemented this for the Questions page: https://github.com/DTS-STN/eligibility-estimator/blob/develop/components/EligibilityPage/index.tsx#L68

That code pulls all questions from the backend. You can use that as a starting point, and then filter out any unanswered questions. This way, you can remove almost all of your duplicated code here with a loop.

Let me know if you have any questions on this!

Reply:
updated